hs_composite_resource/models.py

```diff
@@ -195,35 +626,173 @@ def supports_delete_folder_on_zip(self, original_folder):
+        # get all resource files at full_path and its sub-folders
+        res_file_objects = ResourceFile.list_folder(self, full_path)
 
         # check any logical file associated with the resource file supports deleting the folder
         # after its zipped
         for res_file in res_file_objects:
-            if not res_file.logical_file.supports_delete_folder_on_zip:
-                return False
-
+            if res_file.has_logical_file:
+                if not res_file.logical_file.supports_delete_folder_on_zip:
+                    return False
         return True
 
     def get_missing_file_type_metadata_info(self):
         # this is used in page pre-processor to build the context
-        # so that the landing page can show what metadata items are missing for each logical file
+        # so that the landing page can show what metadata items are missing for each
+        # logical file/aggregation
         metadata_missing_info = []
         for lfo in self.logical_files:
             if not lfo.metadata.has_all_required_elements():
-                file_path = lfo.files.first().short_path
                 missing_elements = lfo.metadata.get_required_missing_elements()
-                metadata_missing_info.append({'file_path': file_path,
+                metadata_missing_info.append({'file_path': lfo.aggregation_name,
                                               'missing_elements': missing_elements})
         return metadata_missing_info
 
+    def update_coverage(self):
+        """Update resource spatial and temporal coverage based on the corresponding coverages
+        from all the contained aggregations (logical file) only if the resource coverage is not
+        already set"""
+
+        # update resource spatial coverage only if it is empty
+        if self.metadata.spatial_coverage is None:
+            self.update_spatial_coverage()
+
+        # update resource temporal coverage only if it empty
+        if self.metadata.temporal_coverage is None:
+            self.update_temporal_coverage()
+
+    def update_spatial_coverage(self):
+        """Updates resource spatial coverage based on the contained spatial coverages of
+        aggregations (file type). Note: This action will overwrite any existing resource spatial
+        coverage data.
+        """
+        spatial_coverages = [lf.metadata.spatial_coverage for lf in self.logical_files
+                             if lf.metadata.spatial_coverage is not None]
+
+        if not spatial_coverages:
+            # no aggregation level spatial coverage data exist - no need to update resource
+            # spatial coverage
+            return
+
+        bbox_limits = {'box': {'northlimit': 'northlimit', 'southlimit': 'southlimit',
+                               'eastlimit': 'eastlimit', 'westlimit': 'westlimit'},
+                       'point': {'northlimit': 'north', 'southlimit': 'north',
+                                 'eastlimit': 'east', 'westlimit': 'east'}
```

Comment:
`southlimit: north` and `westlimit: east`
Is this intentional?  If so, I think a comment is in order.

Reply:
Can they be removed because it's a point?